### PR TITLE
add scraper for fda drugs and aetna drugs (for prescription). Also added flu vaccine

### PR DESCRIPTION
**Pull request by harshitgupta412**

### CODE REVIEW

**Comment by labhinav** [comment]
Does this link with the logic program?

**Comment by harshitgupta412** [comment]
> Does this link with the logic program?

Not yet. FDA Drugs are not yet used in the logic. FDA contraceptives are but rn they are generated in a different file instead of cardinal_care.hrf.

**Comment by labhinav** [comment]
Are there test cases for fda contraceptives?

**Comment by harshitgupta412** [comment]
> Are there test cases for fda contraceptives?

yes

File: claims_analysis/cardinal_care/cardinal_care.hrf [hunk 1]
```diff
@@ -59,67 +63,228 @@ eligible_service(C,P,X):-
...
+% LAIV4 should not be used for persons with compromised immunity 
+eligible_service(C,P,preventive_care):-
+  claim.location(C,L) &
+  valid_location(preventive_care,L) &
+  claim.vaccine(C,flu) &
+  claim.claimant(C,Cl) &
+  person.dob(Cl,DOB) &
+  claim.hosp_start_time(C,C_D,C_T) &
+  get_date_diff_in_months(C_D, DOB, M) &
+  leq(min_age, M) & leq (M, max_age) &
```
Comment (labhinav) [inline claims_analysis/cardinal_care/cardinal_care.hrf:127]: I don't think leq should be used with uninitialized variables, ideally we should initialize them first then run leq to check if they are valid

File: claims_analysis/cardinal_care/cardinal_care.hrf [hunk 2]
```diff
@@ -59,67 +63,228 @@ eligible_service(C,P,X):-
...
+  claim.previous_vaccines_flu_more_than_2(Prev_Count) &
```
Comment (labhinav) [inline claims_analysis/cardinal_care/cardinal_care.hrf:128]: we should update documentation to make it clear that this is annual count

**Review by labhinav — CHANGES_REQUESTED** [review]

File: claims_analysis/cardinal_care/cardinal_care.hrf (hunk 2 above)
Comment (harshitgupta412) [reply claims_analysis/cardinal_care/cardinal_care.hrf:128]: the Prev_count is not annual. It is lifetime.

Comment (harshitgupta412) [reply claims_analysis/cardinal_care/cardinal_care.hrf:128]: vaccine dose count is still annual as per the schema

Comment (labhinav) [reply claims_analysis/cardinal_care/cardinal_care.hrf]: vaccine dose count used elsewhere is not annual, this should be a separate field in that case

File: claims_analysis/cardinal_care/cardinal_care.hrf (hunk 1 above)
Comment (labhinav) [reply claims_analysis/cardinal_care/cardinal_care.hrf]: What are min_age and max_age here?

Comment (harshitgupta412) [reply claims_analysis/cardinal_care/cardinal_care.hrf]: similar to general vaccine clauses. check if the current age is in some range and that range has a corresponding limit.

File: claims_analysis/cardinal_care/cardinal_care.hrf [hunk 3]
```diff
@@ -112,9 +112,26 @@ eligible_service(C,P,preventive_care):-
...
+eligible_service(C,P,preventive_care):-
+  claim.location(C,L) &
+  valid_location(preventive_care,L) &
+  claim.vaccine(C,flu) &
+  claim.claimant(C,Cl) &
+  person.dob(Cl,DOB) &
+  claim.hosp_start_time(C,C_D,C_T) &
+  get_date_diff_in_months(C_D, DOB, M) &
+  claim.previous_vaccines_flu_more_than_2(Prev_Count) & % more than 2 vaccine doses till now (not annual)
+  age_range_limit_flu(C,N,min_age,max_age,Prev_Count) &
```
Comment (labhinav) [inline claims_analysis/cardinal_care/cardinal_care.hrf:128]: Shouldn't min_age and max_age be variables?

Comment (labhinav) [reply claims_analysis/cardinal_care/cardinal_care.hrf:128]: Are all test cases passing?

File: claims_analysis/cardinal_care/cardinal_care.hrf [hunk 4]
```diff
@@ -255,6 +272,11 @@ eligible_for_revaccination(Cl):-
   person.condition(Cl,hemodialysis_patient).
 
 
+%%% Age limits for flu vaccines
+age_range_limit_flu(C,1,0,200,PC).
+age_range_limit_flu(C,2,6,96,no).
```
Comment (labhinav) [inline claims_analysis/cardinal_care/cardinal_care.hrf:277]: Why just 200? A person could be much older than 200 months

**Review by labhinav — CHANGES_REQUESTED** [review]

**Comment by labhinav** [comment]
also can you add person.immunocompromised to schema

**Review by labhinav — APPROVED** [review]

**Comment by labhinav** [comment]
Please create PR to main as well

**Pull request merged**